crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -56,66 +143,193 @@ impl TryFrom<&NovalnetRouterData<&PaymentsAuthorizeRouterData>> for NovalnetPaym
+                    card_expiry_month: req_card.card_exp_month,
+                    card_expiry_year: req_card.card_exp_year,
+                    card_cvc: Some(req_card.card_cvc),
+                    card_holder: item.router_data.get_optional_billing_full_name(),
+                });
+
+                let enforce_3d = match item.router_data.auth_type {
+                    enums::AuthenticationType::ThreeDs => Some(1),
+                    enums::AuthenticationType::NoThreeDs => None,
+                };
+                let test_mode = match item.router_data.test_mode {
+                    Some(true) => 1,
+                    Some(false) | None => 0,
+                };
+
+                let return_url = result_to_option(item.router_data.request.get_return_url());
+                let hook_url = result_to_option(item.router_data.request.get_webhook_url());
+                let transaction = NovalnetPaymentsRequestTransaction {
+                    test_mode,
+                    payment_type: NovalNetPaymentTypes::CREDITCARD,
+                    amount: Some(item.amount.clone()),
+                    currency: Some(item.router_data.request.currency.to_string()),
+                    order_no: Some(item.router_data.connector_request_reference_id.clone()),
+                    hook_url,
+                    return_url: return_url.clone(),
+                    error_return_url: return_url.clone(),
+                    payment_data: novalnet_card,
+                    enforce_3d,
                 };
+
+                let billing = NovalnetPaymentsRequestBilling {
+                    house_no: item.router_data.get_optional_billing_line1(),
+                    street: item.router_data.get_optional_billing_line2(),
+                    city: item.router_data.get_optional_billing_city(),
+                    zip: item.router_data.get_optional_billing_zip(),
+                    country_code: item.router_data.get_optional_billing_country(),
+                };
+
+                let customer_ip = item
+                    .router_data
+                    .request
+                    .get_browser_info()?
+                    .get_ip_address()?;
+
+                let customer = NovalnetPaymentsRequestCustomer {
+                    first_name: result_to_option_secret_string(
+                        item.router_data.get_billing_first_name(),
+                    ),
+                    last_name: item.router_data.get_optional_billing_last_name(),
+                    email: item.router_data.get_optional_billing_email(),
+                    mobile: result_to_option_secret_string(
+                        item.router_data.get_billing_phone_number(),
+                    ),
+                    billing,
+                    customer_ip,
+                };
+
+                let lang = item
+                    .router_data
+                    .request
+                    .get_browser_info()?
+                    .get_language()?;
```

Comment:
Can we pass default language as English if `browser_info` or `language` is not present?

Reply:
sure